### DISCUSSION TITLE
Is it possible on Freeform 3 to use the element connections to add/merge the entry in the element connection and not replace the entry?

### DISCUSSION THREAD

**Discussion by terryupton** [solspace/craft-freeform#936] [Questions & Answers] [answered]
Currently, it looks like it only replaces the entry in the element connections rather than adding to it.
I would like to just add to the current element and not replace it.
Is this possible @kjmartens ?

Thanks
Terry

**Comment by kjmartens** [comment]
Hi @terryupton,

Could you explain more about what you mean by this? 🙂 

To confirm, you're referring to this:
https://docs.solspace.com/craft/freeform/v3/integrations/elements/

And if so, are you generating new elements? Or attempting to edit them, etc?

**Comment by terryupton** [comment]
Hi @kjmartens ,

Yes that is correct (https://docs.solspace.com/craft/freeform/v3/integrations/elements/)
I am attempting to edit existing ones.

Essentially it would be good to either have an option (replace or merge) when assigning element connections and mapping to fields. If the field is an entry field to merge as default and not replace.

My specific situation is as follows: 

A student is enrolling on a new course and they choose their course and then their lesson time/day.
These are both separate entries that live in two separate structures (Tuition Courses and Tuition Lessons).
For transparency, the lessons are related to the courses.

Students are also a structure and are their own entries. 
The student already has an existing course and lesson time/day which are related entries in the student entries.
When I map the newly chosen Tuition Course entry and Lesson entry, these are replacing the existing entries. However, I want them to merge the new entries not replace the existing entries. This is because they are picking the course/time for the next period and their existing course and lesson are still running so they need to remain.

I hope this makes sense, but if not, please shout and I will try and illustrate better with screenshots or a video. 

Thanks
Terry

**Reply by kjmartens** [reply]
Ok, I think I understand what you're saying now. Ultimately, you are having users edit the Tuition Course entry and select another Tuition Lesson entry (which is a Relation field), and you're wanting it to add the new selection to the existing selection instead of replace it? Does that sound right?

If so, this isn't something that is possible right now, be we can note it for future consideration. 🙂

**Reply by terryupton** [reply, marked as answer]
Hey Kelsey,
Bingo! That's exactly it! 
I have managed to circumvent this with a module and events at the moment so it is ok, but it would make life easier in the future for such functionality.

Thank you.
Terry
